Integration test

lmilleri opened the pull request:
Run an e2e test with the sample attester in an ephemeral kind cluster.
This test requires the k8s kuttl plugin and kind installation when executed in local (e.g. make test-e2e).
Also added github action for executing the test on PR and merge in the main branch

lmilleri commented:
PR reshaped to use the kuttl plugin, please review

spotlesstofu commented on README.md:
```diff
@@ -312,6 +312,18 @@ which provide a reconcile function responsible for synchronizing resources until
+
+An attestation with the sample-attester is performed in an ephemeral kind cluster
+Pre-requirements:
+- k8s kuttl plugin installed
```
Please add a link to how to do this.

spotlesstofu commented:
Can you make this run on Github?

bpradipt commented on tests/e2e/sample-attester/create-other-secret.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+oc create secret generic kbsres1 --from-literal key1=res1val1 --from-literal key2=res1val2 -n trustee-operator-system
```
s/oc/kubectl

bpradipt commented on Makefile:
```diff
@@ -172,6 +172,15 @@ build-installer: manifests generate kustomize ## Generate a consolidated YAML wi
+# pre-requirements: k8s kuttl plugin installed
+# you can change the trustee image by defining the env variable KBS_IMAGE_NAME
+# Usage: REGISTRY=<user-registry> KBS_IMAGE_NAME=<trustee-image> make e2e-test
+.PHONY: e2e-test
```
nit: s/e2e-test/test-e2e

lmilleri commented:
Addressed comments and added github action.
Github action can't be tested right now, it'll be active when this PR is merged (tested in my personal repo though).
Please review

spotlesstofu commented on tests/e2e/sample-attester/install-operator.sh:
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+KBS_IMAGE_NAME="${KBS_IMAGE_NAME:-quay.io/confidential-containers/trustee:290fd0eb64ab20f50efbd27cf80542851c0ee17f}"
```
Remove the default value for this variable, we want the test to fail if it's not set.

lmilleri replied:
We'd need both the trustee and the client images to be specified and they are mandatory for the test to execute. I'll do the changes for the client image customization.
I think the default images have to be provided somehow, otherwise the github action won't be able to execute properly. Where do you reckon these images should be hardcoded?
1) github action: ok for CI but if you run `make test-e2e` the images must be provided in the command
2) Makefile: both CI and `make test-e2e` are ok

spotlesstofu replied:
Makefile
